src/rules/noSelfAssignRule.ts

```diff
@@ -0,0 +1,210 @@
+/**
+ * @license
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isArrayLiteralExpression, isBinaryExpression, isElementAccessExpression, isIdentifier,
+    isLiteralExpression, isObjectLiteralExpression, isPropertyAccessExpression, isPropertyDeclaration,
+    isShorthandPropertyAssignment, isSpreadAssignment, isSpreadElement } from "tsutils";
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+interface Option {
+    props: boolean;
+}
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "no-self-assign",
+        description: "This rule is aimed at eliminating self assignments.",
```

Comment:
This description doesn't add anything. "This rule is aimed at" is implied; "eliminating" is a longer form of "no"; "assignments" is abbreviated by "assign". Amusing that that's what the ESLint docs call it, but can we get a better description here? Maybe:

> Stops variables from being assigned to themselves.